optimum/exporters/tasks.py

```diff
@@ -1121,10 +1121,11 @@ def decorator(config_cls: Type) -> Type:
                 mapping = supported_model_type_for_library.get(model_type, {})
                 mapping_backend = mapping.get(backend, {})
                 for task in supported_tasks:
-                    if task not in cls.get_all_tasks():
+                    normalized_task = task.replace("-with-past", "")
+                    if normalized_task not in cls.get_all_tasks():
```

Comment:
Is this change related to the refactor?

Reply:
it is not related to refactoring, it fixes bug that prevent registering tasks with-past in this case (I found it testing on optimum-intel side for registering configs for openvino)